cEP-0014.md

```diff
@@ -0,0 +1,324 @@
+        ]
+}
+```
+
+The mappings will be stored in a module named `InfoMappings.py` in quickstart inside the `info_extraction` package.
+
+#### Filling-in the values of bear settings
+
+An attempt to autofill the setting's value can be made before prompting the user to fill the values. This can be done as follows:
+```
+def autofill_value_if_possible(self,
+                               setting_name,
+                               section_name,
+                               bear_name):
+    if INFO_SETTING_MAPS.get(setting_name):
+        for mapping in INFO_SETTING_MAPS[setting_name]:
+            scope = mapping["scope"]
+            if (scope.check_belongs_to_scope(
+                    section_name, bear_name):
+                # look for the values in extracted information
+                # from all the ``InfoExtractor`` instances.
+                values = extracted_information.get(mapping["info_kind"])
+                for val in values:
+                    if scope.check_is_applicable_information(val):
+                        yield val
+    return None
+```
+
+
+#### Detecting inconsistencies in project configurations
+Another interesting application of the unified representation of extracted-information is detecting the inconsistencies among different project configurations. The intra-scope conflicts due to different Info from different sources can be brought up to users consideration. The inter-scope conflicts can be fixed either based on level-priority (global < section < bear) or by prompting users to select their preferred value.
+
+The inconsistencies can be detected as
+```
+to_fill_values = list(autofill_value_if_possible(some_setting_name,
+                                                 some_section_name,
+                                                 some_bear_name))
+if len(to_fill_values) > 1:
+    # warn the user about inconsistency.
+```
+
+The inconsistencies need to be resolved before using the `Info` values to fill in the settings of coafile sections. As a simple solution, user can be prompted like
+```
+> We found conflicting options in your file_a and your file_b, which should assume dominance?
+1. file_a
+2. file_b
+3. override both and specify your value
+```
+
+
+### Overall effect on coala-quickstart's CLI and outputs
+After the bears are filtered, and relevant information has been extracted out of the projects, the users are notified, a more tailored version of coafile is written.
+The per-language sections are generated like before, but with some additional bears included based on relevant information discovered from project and user's preference (CLI input).
+The users are also prompted for the values like before, but an attempt is made to autofill these values before prompting the users. If there are multiple possible values to autofill, such inconsistencies are brought to user's consideration allowing him to pick any one or them or override all of them.
+
+#### Running mode options for quickstart
```

Comment:
@hemangsk running Non-interactive mode without any predefined capabilities will activate all the possible bears based on the language.